plugin/src/sbt-test/sbt-jni/simple-cargo/build.sbt

```diff
@@ -3,7 +3,11 @@ ivyLoggingLevel := UpdateLogging.Quiet
 lazy val root = (project in file(".")).aggregate(core, native)
 
 lazy val core = project
-  .settings(libraryDependencies += "org.scalatest" %% "scalatest" % "3.2.9" % Test)
+  .settings(
+    libraryDependencies += "org.scalatest" %% "scalatest" % "3.2.9" % Test,
+    sbtJniCoreScope := Compile,
+    classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.Flat // otherwise you get java.lang.UnsatisfiedLinkError: 'int simplecargo.Adder.plus(int)'
```

Comment:
Just curious, why it worked before? Or it happens only with Cargo when the `syntax.NativeLoader` is used? I use it [here](https://github.com/PDAL/java/blob/main/core/src/main/scala/io/pdal/Pipeline.scala#L44) and see no errors.

Reply:
> happens only with Cargo
nn, not really. I've able to reproduce it on other `sbt-test` projects too, like `simple-syntax`. So maybe I shouldn't be changing the `classLoaderLayeringStrategy` only here -- either everywhere where necessary or nowhere.

I manually launched `sbt` in `plugin/src/sbt-test/sbt-jni/simple-syntax` and manually executing the steps (descibed in the file `test`).

Are you able to reproduce the `UnsatisfiedLinkError` on your machine or not?

> I use it here and see no errors

Have you tried running/testing that code from sbt?

Reply:
@sideeffffect what code you mean? but yes, [test](https://github.com/PDAL/java/blob/main/core/src/test/scala/io/pdal/PipelineSpec.scala) are [launched](https://github.com/PDAL/java/blob/main/.github/workflows/ci.yml#L31)

What are the steps to reproduce the `UnsatisfiedLinkError`?

Reply:
Also this project CI runs tests from sbt (via `sbt test-plugin`) and tests pass there. https://github.com/sbt/sbt-jni/runs/4613550346?check_suite_focus=true 

```
[info] Running 5 / 5 (100.00%) scripted tests with RunFromSourceMain
[info] Running sbt-jni/multiclasses
[info] 
[info] Running sbt-jni/oneproject
[info] 
[info] Running sbt-jni/simple
[info] 
[info] Running sbt-jni/simple-cargo
[info] 
[info] Running sbt-jni/simple-syntax
```

Reply:
1. tried `sbt test-plugin` locally and tests pass.
2. manual execution from the folder:

```bash
$ pwd
/.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo
[simple-cargo (main)]$ sbt -Dplugin.version=1.5.3+25-1aa1902d-SNAPSHOT
[info] welcome to sbt 1.5.8 (Eclipse Foundation Java 11.0.12)
[info] loading global plugins from /Users/.../.sbt/1.0/plugins
[info] loading settings for project simple-cargo-build from plugins.sbt ...
[info] loading project definition from /.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/project
[info] loading settings for project root from build.sbt ...
[info] set current project to root (in build file:/.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/)
[info] sbt server started at local:///.../.sbt/1.0/server/d6509e5bbd7bac406f1a/sock
[info] started sbt server
sbt:root> nativeInit cargo adder
[info] Initializing native build with Cargo configuration
[info] Wrote to /.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/native/Cargo.toml
[success] Total time: 0 s, completed Dec 22, 2021, 9:38:11 PM
sbt:root> +test
[info] Setting Scala version to 2.12.14 on 3 projects.
[info] Reapplying settings...
[info] set current project to root (in build file:/.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/)
[info] Building library with native build tool Cargo
[warn]     Updating crates.io index
[info] compiling 2 Scala sources to /.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/core/target/scala-2.12/classes ...
[warn]    Compiling proc-macro2 v1.0.28
[warn]    Compiling unicode-xid v0.2.2
[warn]    Compiling syn v1.0.74
[warn]    Compiling same-file v1.0.6
[warn]    Compiling memchr v2.4.0
[warn]    Compiling log v0.4.14
[warn]    Compiling cfg-if v1.0.0
[warn]    Compiling bytes v1.0.1
[warn]    Compiling cesu8 v1.1.0
[warn]    Compiling jni-sys v0.3.0
[warn]    Compiling walkdir v2.3.2
[warn]    Compiling jni v0.19.0
[warn]    Compiling quote v1.0.9
[warn]    Compiling combine v4.6.0
[warn]    Compiling thiserror-impl v1.0.26
[warn]    Compiling thiserror v1.0.26
[warn]    Compiling adder v0.1.0 (/.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/native)
[warn]     Finished release [optimized] target(s) in 50.29s
[success] Library built in /.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/native/target/native/x86_64-darwin/bin/release/libadder.dylib
[info] compiling 1 Scala source to /.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/core/target/scala-2.12/test-classes ...
Printing from rust library. base: 12
Printing from rust library. term: 34
[info] SimpleSpec:
[info] Calling native methods in tests
[info] - should work
[info] Run completed in 433 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 52 s, completed Dec 22, 2021, 9:39:07 PM
[info] Setting Scala version to 2.13.7 on 3 projects.
[info] Reapplying settings...
[info] set current project to root (in build file:/.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/)
[info] Building library with native build tool Cargo
[warn]     Finished release [optimized] target(s) in 0.05s
[success] Library built in /.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/native/target/native/x86_64-darwin/bin/release/libadder.dylib
[info] compiling 1 Scala source to /.../pomadchin/sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/core/target/scala-2.13/test-classes ...
Printing from rust library. base: 12
Printing from rust library. term: 34
[info] SimpleSpec:
[info] Calling native methods in tests
[info] - should work
[info] Run completed in 392 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 5 s, completed Dec 22, 2021, 9:39:12 PM
[info] Reapplying settings...
[info] set current project to root (in build file:/.../sbt-jni/plugin/src/sbt-test/sbt-jni/simple-cargo/)
sbt:root> 
```

Reply:
Ah, very strange, but now I see!
It's been caused by
```
ThisBuild / turbo := true
```
in my `~/.sbt/1.0/build.sbt` :facepalm:

Reply:
@sideeffffect do you still want to cleanup this PR to get it merged?

Reply:
Yes, I'll get to it tomorrow. :sleeping_bed:  Thanks for now! :heart:

Reply:
@sideeffffect thanks! 🚀